LSPS2/README.md

```diff
@@ -0,0 +1,864 @@
+where the next hop is `jit_channel_scid`, before
+`opening_fee_params.valid_until`:
+
+* MUST set `payment_size_msat` as the forwarded value of the HTLC.
+* MUST compute the `opening_fee` based on the `payment_size_msat`, and
+  if the computation overflows, MUST fail with `unknown_next_peer`.
+* MUST check that `opening_fee + htlc_minimum_msat < payment_size_msat`,
+  and if that fails, MUST fail with `unknown_next_peer`.
+
+### 5.  Channel Opening And Forwarding
+
+The LSP requests a channel open to the client.
+
+The LSP selects the channel size.
+The LSP MUST ensure the channel is large enough to transmit the
+`payment_size_msat - opening_fee`, plus any reserve needed on the LSP
+side.
+
+The LSP and client MUST negotiate these options and parameters:
+
+* `option_zeroconf` is set.
+* `option_scid_alias` is set.
+* `announce_channel` is `false`.
+* The client-provided `to_self_delay` is less than or equal to
+  the agreed-upon `max_client_to_self_delay`.
+
+> **Rationale** `option_zeroconf` is set so that the payment can
+> be immediately forwarded to the client and resolved as soon as
+> the opening negotiation completes; if the client and LSP had to
+> wait for confirmation, the incoming payment would have to be
+> locked for an extended period, which is indistinguishable from
+> a channel jamming attack.
+> `option_zeroconf` cannot be set unless `announce_channel` is
+> `false`.
+> `option_scid_alias` needs to be set so that the channel can be
+> referred to on future invoices before the channel confirms.
+>
+> Future versions of this API may be able to utilize Asynchronous
+> Receive (currently being developed as of this version) to
+> be able to get around the `option_zeroconf` requirement, by
+> treating the receiver / client as offline until the client is
+> online *and* the JIT channel to it is deeply confirmed, if the
+> LSP and client negotiate a non-`option_zeroconf` channel for
+> the JIT channel.
+> A non-`option_zeroconf` JIT channel could also be later
+> published with `announce_channel = true`.
+
+Clients MAY reject the channel open attempt if any other
+parameters or options are not desired by the client (for example,
+it MAY reject non-zero channel reserve requirements imposed on
+the client by the LSP).
+Rejection is via the `error` Lightning BOLT message.
+
+LSPs MUST fail the payment with `unknown_next_peer` in case the
+client rejects the channel open via an `error`.
+
+If the client disconnects before the LSP can receive `funding_signed`
+from the client, the LSP MUST fail the incoming payment parts with
+`temporary_channel_failure`, then return to the previous step (waiting
+for payment).
+
+> **Rationale** The client might have crashed at this point, and
+> recovery might take an inordinate amount of time.
+> However, use of the `temporary_channel_failure` error informs the
+> payer that the error is temporary and it can retry later, if the
+> client is online at that point.
```

Comment:
So we fail the payment but still open the channel later? Doesn't this open an attack possibility?

Reply:
Client disconnection before `funding_signed` means the channel open fails --- the LSP node should not broadcast its funding transaction and if it manages its onchain funds the onchain funds should now be returned to an "allocatable" state for future opens.  If it does not, then the LSP node software would have a monetary-loss bug.

Since the channel open failed, we have to fail the payment.